### PR TITLE
helm: add deployment annotation

### PR DESCRIPTION
**Pull request by krrrr38**
Signed-off-by: krrrr38 <k.kaizu38@gmail.com>

### Issue

Some crd / tools requires deployment annotations.

### Description

Add deployment annotation values into aws-load-balancer-controller helm chart.

### Checklist
- [ ] ~~Added tests that cover your change (if possible)~~
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] ~~Backfilled missing tests for code in same general area :tada:~~
- [ ] ~~Refactored something and made the world a better place :star2:~~

### Testing

with annotations

```sh
> helm template . | \grep -A10 -B3 "kind: Deployment"
---
# Source: aws-load-balancer-controller/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: RELEASE-NAME-aws-load-balancer-controller
  namespace: default
  annotations:
    a1: b1
    a2: b2
  labels:
    helm.sh/chart: aws-load-balancer-controller-1.4.1
    app.kubernetes.io/name: aws-load-balancer-controller
    app.kubernetes.io/instance: RELEASE-NAME
```

without annotations

```sh
> helm template . | \grep -A10 -B3 "kind: Deployment"
---
# Source: aws-load-balancer-controller/templates/deployment.yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: RELEASE-NAME-aws-load-balancer-controller
  namespace: default
  labels:
    helm.sh/chart: aws-load-balancer-controller-1.4.1
    app.kubernetes.io/name: aws-load-balancer-controller
    app.kubernetes.io/instance: RELEASE-NAME
    app.kubernetes.io/version: "v2.4.1"
    app.kubernetes.io/managed-by: Helm
spec:
```

### CODE REVIEW

File: helm/aws-load-balancer-controller/values.yaml [hunk 1]
```diff
@@ -91,6 +91,8 @@ updateStrategy: {}
 # serviceAnnotations contains annotations to be added to the provisioned webhook service resource
 serviceAnnotations: {}
 
+deploymentAnnotations: {}
```
Comment (kishorj) [inline helm/aws-load-balancer-controller/values.yaml:94]: Please add a comment for this value, something like:
```
# deploymentAnnotations contains additional annotations for the controller deployment
deploymentAnnotations: {}
```

**Comment by kishorj** [comment]
Also update the README.md file for the new value

File: helm/aws-load-balancer-controller/values.yaml (hunk 1 above)
Comment (krrrr38) [reply helm/aws-load-balancer-controller/values.yaml:95]: Added comment, test.values and REAME.md

https://github.com/kubernetes-sigs/aws-load-balancer-controller/pull/2611/commits/e2fe0424383fc7aaae08d798ce64d684ff989335

**Pull request merged**